### PR TITLE
cli: cluster config: tenant name may not contain hyphen

### PR DESCRIPTION
**Pull request by MoSattler**
Addresses [this](https://github.com/opstrace/opstrace/issues/710#issuecomment-871300624).

# Have you...

* [x] Discussed your change with a project contributor in an issue yet?
* [x] Added an explanation of what your changes do?
* [x] Written new tests for your changes?
* [x] Thought about which docs need updating?

### CODE REVIEW

File: packages/cli/src/schemasv2.test.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+        it("name may not contain special characters", async () => {
+            const configFile = generateValidClusterConfigFile({ tenants: ["abc^def"] })
+            expect(await ClusterConfigFileSchemaV2.isValid(configFile)).toBe(false)
+        })
+
+        it("name may not contain hyphen", async () => {
+            const configFile = generateValidClusterConfigFile({ tenants: ["my-tenant"] })
+            expect(await ClusterConfigFileSchemaV2.isValid(configFile)).toBe(false)
+        })
```
Comment (MoSattler) [inline packages/cli/src/schemasv2.test.ts:79]: Not all of them reflex the *desired* behaviour - but it does reflex the current behaviour. 

Meaning some of those case might be in fact bugs.

File: packages/config/src/clusterconfig.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ export const CLUSTER_NAME_REGEX = /^[a-z0-9-_]+$/;
 
 // This is the same regex the k8s ecosystem uses for dns-related label values
 // see https://github.com/opstrace/opstrace/issues/710
```
Comment (MoSattler) [inline packages/config/src/clusterconfig.ts:32]: I guess this comment is outdated now - should I edit/remove it?

Comment (jgehrcke) [reply packages/config/src/clusterconfig.ts:32]: maybe edit it! leave the reference to issue 710 there and add another reference explaining why we also have to remove the dash for now.

**Comment by jgehrcke** [comment]
:rocket:

**Pull request merged**